pkg/controller/contrailcni/job.go

```diff
@@ -0,0 +1,212 @@
+package contrailcni
+
+import (
+	batch "k8s.io/api/batch/v1"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CniDirs is a struct with data deciding which directories to cover with cni configurtion
+type CniDirs struct {
+	BinariesDirectory string
+	DeploymentType    string
+}
+
+// GetJob is a method that returns k8s Job object filled with containers configuration contrail CNI plugin
+func GetJob(cniDir CniDirs, requestName, instanceType string, replicas *int32) *batch.Job {
+	var trueVal = true
+
+	var cniContainer = core.Container{
+		Name:  "vroutercni",
+		Image: "hub.juniper.net/contrail-nightly/contrail-kubernetes-cni-init:master.latest",
+		Command: []string{"sh", "-c",
+			"mkdir -p /host/etc_cni/net.d && " +
+				"mkdir -p /var/lib/contrail/ports/vm && " +
+				"cp -f /usr/bin/contrail-k8s-cni /host/opt_cni_bin && " +
+				"chmod 0755 /host/opt_cni_bin/contrail-k8s-cni && " +
+				"cp -f /etc/contrailconfigmaps/10-contrail.conf /host/etc_cni/net.d/10-contrail.conf && " +
+				"tar -C /host/opt_cni_bin -xzf /opt/cni-v0.3.0.tgz"},
+		VolumeMounts: []core.VolumeMount{
+			{
+				Name:      "var-lib-contrail",
+				MountPath: "/var/lib/contrail",
+			},
+			{
+				Name:      "cni-config-files",
+				MountPath: "/host/etc_cni",
+			},
+			{
+				Name:      "cni-bin",
+				MountPath: "/host/opt_cni_bin",
+			},
+			{
+				Name:      "var-log-contrail-cni",
+				MountPath: "/host/log_cni",
+			},
+			{
+				Name:      "vrouter-logs",
+				MountPath: "/var/log/contrail",
+			},
+			{
+				Name:      "configmap-volume",
+				MountPath: "/etc/contrailconfigmaps",
+			},
+			{
+				Name:      "etc-kubernetes-cni",
+				MountPath: "/etc/kubernetes/cni",
+			},
+			{
+				Name:      "multus-cni",
+				MountPath: "/var/run/multus",
+			},
+		},
+		ImagePullPolicy: "Always",
+		SecurityContext: &core.SecurityContext{
+			Privileged: &trueVal,
+		},
+	}
+
+	if cniDir.DeploymentType == "openshift" {
+		cniContainer.Command[len(cniContainer.Command)-1] = cniContainer.Command[len(cniContainer.Command)-1] +
+			"&& mkdir -p /etc/kubernetes/cni/net.d && " +
```

Comment:
Extra space is required here " && mkdir ..."

Reply:
Done